src/main/java/org/isf/lab/manager/LabManager.java

```diff
@@ -265,6 +265,28 @@ public boolean newLaboratory2(Laboratory laboratory, List<LaboratoryRow> labRow)
 		}
 	}
 
+	/**
+	 * Inserts one Laboratory exam {@link Laboratory} (All Procedures)
+	 *
+	 * @param laboratory - the laboratory without result
+	 * @return <code>true</code> if the exam has been inserted, <code>false</code> otherwise
+	 * @throws OHServiceException
+	 */
+	public boolean newExamRequest(Laboratory laboratory) throws OHServiceException {
+		setPatientConsistency(laboratory);
+		return ioOperations.newLabFirstProcedure(laboratory);
+	}
+	
+	/**
+	 * update one Laboratory exam {@link Laboratory} (All Procedures)
+	 *
+	 * @param laboratory - the laboratory without result
+	 * @return <code>true</code> if the exam has been update, <code>false</code> otherwise
+	 * @throws OHServiceException
+	 */
+	public boolean updateExamRequest(Laboratory laboratory) throws OHServiceException {
+		return ioOperations.updateLabFirstProcedure(laboratory);
+	}
```

Comment:
in terms of API, wouldn't be better to have a "new" method that forces the state to 'draft' and another method that just update the state?

If I want to update the laboratory (with results) there are already other methods.

Also the javadoc should be changed accordingly

Reply:
yes if you want to update the laboratory with results we already have the method for that case.  about new method that force the state to 'draft' and another method that just update the state now if you want to update the state with another fields you will use other method while you can just use the updateExamRequest for all case.

Reply:
What I mean is that `newExamRequest` should call `laboratory.setStatus("draft")` before the ioOperations and the result should be blank even if passed, otherwise one could use the same method to save a laboratory first procedure with result directly in "done" and so duplicating the `newLabFirstProcedure` method.

Similarly, the method `updateExamRequest` should only change the laboratory status and the result should be unchanged, otherwise it may be used to update a laboratory first procedure and its result, duplicating the `updateLabFirstProcedure` method: in this case we could accept `code` and `status` as the only parameters.

In other words, a well-defined method should perform only the expected action, and the javadoc also should be more precise in this.

Also, a laboratoryStatus `enum` should be defined and used here.